x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/SettingsValidationContext.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan;
+
+import org.elasticsearch.Build;
+import org.elasticsearch.transport.RemoteClusterService;
+
+public class SettingsValidationContext {
```

Comment:
nit: As the tests are subclassing this, and to avoid calling `Build.current().isSnapshot()` multiple times in production, maybe this should be a record (crossProjectEnabled, isSnapshot), and have a custom constructor accepting the RemoteClusterService and calling isSnapshot().
Maybe this approach is somewhat shortsighted if we want to add more derived states or services, so take it with a grain of salt!